File: domain/src/main/java/ch/threema/domain/protocol/api/work/WorkContact.java
package ch.threema.domain.protocol.api.work;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class WorkContact {
    public final @Nullable String firstName;
    public final @Nullable String lastName;
    public final @NonNull byte[] publicKey;
    public final @NonNull String threemaId;
    public final @Nullable String jobTitle;
    public final @Nullable String department;

    public WorkContact(
        @NonNull String threemaId,
        @NonNull byte[] publicKey,
        @Nullable String firstName,
        @Nullable String lastName,
        @Nullable String jobTitle,
        @Nullable String department
    ) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.publicKey = publicKey;
        this.threemaId = threemaId;
        this.jobTitle = jobTitle;
        this.department = department;
    }
}

File: domain/src/main/java/ch/threema/domain/protocol/api/work/WorkContact.kt
package ch.threema.domain.protocol.api.work

import java.time.Instant

open class WorkContact(
    @JvmField val threemaId: String,
    @JvmField val publicKey: ByteArray,
    @JvmField val firstName: String?,
    @JvmField val lastName: String?,
    @JvmField val jobTitle: String?,
    @JvmField val department: String?,
    @JvmField val availability: String?,
    @JvmField val workLastFullSyncAt: Instant?,
) {
    override fun toString(): String =
        "WorkContact(threemaId=$threemaId, publicKey=${publicKey.contentToString()}, firstName=$firstName, lastName=$lastName, " +
            "jobTitle=$jobTitle, department=$department, availability=$availability, workLastFullSyncAt=$workLastFullSyncAt)"
}
